swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+from swap_meet.item import Item 
+"""
+Module that contains class Vendor with attribute inventory and instance methods: add, remove.
+Other instance methods:
+To get items in inventory by Item object attribute: get_by_id, get_by_category, get_best_by_category, get_newest_by_category 
+To swap items by attribute: swap_best_by_category, , swap_newest_by_category. 
+"""
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+        
+    def add(self, item):
+        self.inventory.append(item)
+        if item in self.inventory:
+            return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        if item not in self.inventory:
+            return item
+    
+    def get_by_id(self, search_id):
+        for item in self.inventory:
+            if item.id == search_id:
+                return item
+        return None   
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not (my_item in self.inventory and their_item in other_vendor.inventory):
+            return False   
+        #Add their item to my inventory, add my item to their inventory 
+        self.inventory.append(their_item)
+        other_vendor.inventory.append(my_item)
+        #Remove their item from their inventory, remove my item from my inventory
+        self.inventory.remove(my_item)   
```

Comment:
I like this way of checking if `inventory` is empty/Falsey

This is a really small piece of knowledge: It's very subtle, but if you change this conditional, you can achieve ["short-circuiting"](https://en.wikipedia.org/wiki/Short-circuit_evaluation). (Sorry if this has already been covered in class and I didn't know.) In compound conditionals with an `and`, if the first part of the conditional (before the `and`) is already `False`, then the program won't even get to the second part after the `and`, it'll know that the overall conditional is `False`. Similarly, if it's an `or`, if the first part is `True`, then the overall conditional is `True`. If we were to apply it here, we'd need to refactor your `if` to:

```python
        if not self.inventory or not other_vendor.inventory:
            return False
```

In the grand scheme of things, in this project, there's not really an argument to refactor it, I just wanted to use the chance to explain something cool :D . There are random cases where this kind of "trick" is useful, though, but I've seen that like once or twice in my career.